test/cli_test.js

```diff
@@ -21,6 +23,40 @@ describe('cli', function() {
     process.chdir(oldCwd);
   });
 
+  describe('build', function() {
+    let exitStub;
+
+    beforeEach(function() {
+      exitStub = sinon.stub(process, 'exit');
+    });
+
+    afterEach(function() {
+      rimraf.sync('dist');
+    });
+
+    context('on successful build', function() {
+      it('cleanups tmp files', function(done) {
+        const cleanup = sinon.spy(Builder.prototype, 'cleanup');
+
+        cli(['node', 'broccoli', 'build', 'dist']);
+
+        process.nextTick(() => {
+          chai.expect(cleanup).to.be.calledOnce;
```

Comment:
If this assertion fails, i believe this test wont fail with a useful error message, rather it will time out.
Remember in mocha, assertions are thrown exceptions. That means the `done` will never be hit...

We will need to address that before we can merge this.

One option would be:

```js
process.nextTick(() => {
  try {
    chai.expect(cleanup).to.be.calledOnce;
    done();
  } catch(e) {
    done(e); // fail the test with the given error
  }
});
```

Reply:
> process.nextTick(() => {

It's a bummer we need to do this, but commander doesn't make this stuff very easy to test.

Reply:
it's super interesting to me now as I you're 100% right with try/catch solution. so I have red/green tested this code by replacing `calledOnce` with `calledTwice` and it gave me perfectly fine error message
```
  1) cli build on successful build cleanups tmp files:                                                                                                                                        
     Uncaught expected cleanup to have been called exactly twice, but it was called once                                                                                                      
    cleanup() at builder.build.then.finally (/home/chrmod/Projects/chrmod/broccoli/lib/cli.js:48:32)   
```
so I've remove `done()` call, and it still worked.

Does mocha somehow handles `process.nextTick`? (`done` argument still have to be present for it to work).

Reply:
there is a mocha test case that does the same https://github.com/mochajs/mocha/blob/2bb2b9fa35818db7a02e5068364b0c417436b1af/test/integration/fixtures/uncaught-hook.fixture.js#L10

guess it is fine to keep it like that

Reply:
interesting, guess they made this work.